When compiling GNAT, ADA will undef `__linux__` if `__ANDROID__` is defined.
So, check both `__ANDROID__` and `__linux__` in libcxx's `__config` header.

--- a/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/__config
+++ b/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/__config
@@ -927,6 +927,7 @@
         defined(__OpenBSD__) ||                                                                                        \
         defined(__NuttX__) ||                                                                                          \
         defined(__linux__) ||                                                                                          \
+        defined(__ANDROID__) ||                                                                                        \
         defined(__GNU__) ||                                                                                            \
         defined(__APPLE__) ||                                                                                          \
         defined(__sun__) ||                                                                                            \
